perma_web/api/resources.py

```diff
@@ -481,11 +481,11 @@ def obj_update(self, bundle, skip_errors=False, **kwargs):
                 going_private = bundle.data.get("is_private")
                 # if link was private but has been marked public
                 if is_private and not going_private:
-                    run_task(upload_to_internet_archive.s(link_guid=bundle.obj.guid))
+                    run_task(upload_to_internet_archive.s(link=bundle.obj))
```

Comment:
Passing `link` instead of `link_guid` won't work for production, where RUN_TASKS_ASYNC=True. This line raises `EncodeError: <Link: 2HWD-BUB9> is not JSON serializable`. To pass tasks to background celery the parameters have to be JSON serializable.

(It would be neat if unit tests found this issue -- not sure if there's a good way to do that though.)

(Even if we used the `pickle` serializer which lets you pass Django objects directly, doing it this way raises the possibility that the link object will be updated before the job is actually run, and the task won't see the updates. It's hard to figure out whether that's a problem in a given case -- simpler just to fetch an up-to-date copy when the task starts.)